lib/utils/emitDeprecationWarning.cjs

```diff
@@ -0,0 +1,28 @@
+// NOTICE: This file is generated by Rollup. To modify it,
+// please instead edit the ESM counterpart and rebuild with Rollup (npm run build).
+'use strict';
+
+const process = require('node:process');
+
+// Each code must be unique.
+const CODES = {
+	COMMONJS_PLUGINS: '001',
+	COMMONJS_NODEJS_API: '002',
+	RESULT_OUTPUT_PROPERTY: '003',
+};
+
+/**
+ * Use this function for deprecation warnings, instead of `process.emitWarning()`.
+ *
+ * @param {string} message
+ * @param {keyof CODES} codeKey
+ * @param {string} detail
```

Comment:
@ybiquitous we could possibly share [`DeprecatedInfo`](https://jsr.io/@eslint/core@0.11.0/doc/~/DeprecatedInfo) with ESLint.

Reply:
I'm not sure about the type of ESLint, but is it possible to pass such more complex info to `process.emitWarning`?

I'm okay with discussing a new issue if there's feasibility.